crates/ruff_linter/resources/test/fixtures/wemake_python_styleguide/control_var_used_after_block.py

```diff
@@ -0,0 +1,90 @@
+for global_var in []:
```

Comment:
> What's the motivation for limiting the rule to `with` and `for` only? Should the rule apply to all compound statements?
>
> *-- @MichaReiser, https://github.com/astral-sh/ruff/pull/11769#pullrequestreview-2196737675*

Ideally, I would like to lint for full block-scoping in Python.

Since Python isn't block-scoped, people often write code with patterns that go against this so there would be a lot of violations in existing codebases for full block-scoping. Using a loop-var after the block is *probably* a mistake though and is a good thing to watch out for in any case (good introduction).

But probably the main motivation for why only `with`/`for` is just to match what the `wemake-python-styleguide` did (prior art). Given the "drop-in" nature that `ruff` seems to try to sit in, I thought I would match this.

> @AlexWaygood mentioned that there reasons to access variables from **inside** a with block. So maybe we should exclude with statements.

Inside should be fine. I assume you typo'ed and meant `ouside` there?

I guess it would be good to know the reasons/use cases to judge better but there are use cases even for the for-loop behavior but the goal is just to lint against that regardless in favor of refactoring to a better pattern.

And if we don't apply this to `with` loops, it's not a drop-in replacement for the other rule. But perhaps that's not a strict goal, etc